tests/integ/test_large_query_breakdown.py

```diff
@@ -430,3 +432,39 @@ def test_large_query_breakdown_enabled_parameter(session, caplog):
     with caplog.at_level(logging.WARNING):
         session.large_query_breakdown_enabled = True
     assert "large_query_breakdown_enabled is experimental" in caplog.text
+
+
+@pytest.mark.skipif(IS_IN_STORED_PROC, reason="requires graphviz")
+@pytest.mark.parametrize("enabled", [False, True])
+def test_plotter(session, large_query_df, enabled):
+    original_plotter_enabled = os.environ.get(
+        "ENABLE_SNOWFLAKE_OPTIMIZATION_PLAN_PLOTTING"
+    )
+    try:
+        os.environ["ENABLE_SNOWFLAKE_OPTIMIZATION_PLAN_PLOTTING"] = str(enabled)
+        tmp_dir = tempfile.gettempdir()
+
+        with patch("graphviz.Graph.render") as mock_render:
+            large_query_df.collect()
+            assert mock_render.called == enabled
+            if not enabled:
+                return
+
+            assert mock_render.call_count == 4
+            expected_files = [
+                "original_plan",
+                "cte_optimized_plan_0",
+                "large_query_breakdown_plan_0",
+                "large_query_breakdown_plan_1",
+            ]
+            for i, file in enumerate(expected_files):
+                path = os.path.join(tmp_dir, "snowpark_query_plan_plots", file)
+                assert mock_render.call_args_list[i][0][0] == path
+
+    finally:
+        if original_plotter_enabled is not None:
+            os.environ[
+                "ENABLE_SNOWFLAKE_OPTIMIZATION_PLAN_PLOTTING"
```

Comment:
shall we also clean up the plot under the dir?

Reply:
we mocked graphviz render so this test does not actually write graphs to directory. 

`with patch("graphviz.Graph.render") as mock_render:`